### PR TITLE
feat(core):add S3 Bucket RemovalPolicy to remove the bucket completely

### PR DESCRIPTION
**Pull request by guan840912**
#7

*Description of changes:*
add s3 Bucket RemovalPolicy in index.ts line 69-72
```typescript
// S3 bucket to host K3s token + kubeconfig file 
    const k3sBucket = new s3.Bucket(this, 'k3sBucket', {
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });
```
let s3 can be destroy when `cdk destroy -f ${stack_name}`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mreferre** [comment]
Thanks Neil!. BTW have you tested this? I am asking because I vaguely remember that the destroy policy only work (worked?) for empty buckets. Given that this bucket is not empty at time of destroy (because it still contains the config yaml and k3s secret) it did not delete the bucket and hence I just removed the policy. If you tested it and it works I am happy to merge the PR (thanks!)

**Comment by guan840912** [comment]
Hi @mreferre ,
 I already add custom resource to delete Object in s3 Bucket , and then 
`removalPolicy: cdk.RemovalPolicy.DESTROY` will be work . ☺️

**Comment by guan840912** [comment]
```bash
# when i run 
cdk destroy -f 
```
截圖 2020-09-12 下午10 36 11
Can see S3 Bucket will be deleted 😊
截圖 2020-09-12 下午10 36 29

File: test/cluster.test.ts [hunk 1]
```diff
@@ -21,4 +21,11 @@ test('create the default cluster', () => {
...
   expect(stack).toHaveResource('AWS::AutoScaling::LaunchConfiguration')
-})
+});
+
+test('add s3 removalPolicy', () => {
+  const app = new App();
+  const stack = new Stack(app, 'testing-stack');
+  new k3s.Cluster(stack, 'Cluster-s3-removalPolicy')
+  expect(stack).toHaveResource('AWS::S3::Bucket');
+});
```
Comment (pahud) [inline test/cluster.test.ts:31]: I was wondering if we should introduce a new property in `ClusterProps` for example `bucketRemovalPolicy?: RemovalPolicy` and the default value is `RemovalPolicy.RETAIN`. When we specify the value as `RemovalPolicy.DESTROY`,  the custom resource removes the s3 bucket completely like this. WDYT @mreferre ? Does it make sense to retain the bucket as default behavior?

Comment (mreferre) [reply test/cluster.test.ts:31]: I like the idea of adding a property to control this behavior. The bucket only contains the the k3s secret and the config.yaml to connect to the cluster and once the cluster is gone the value of those two files is close to 0. Having that said I don't like the idea to delete the bucket by default. I'd rather document front and center this property and call out that the default is to leave the bucket behind but users can opt in and decide to delete it if so they wish.

**Review by pahud — APPROVED** [review]
LGTM. 

@mreferre I am fine with this PR.

**Comment by mreferre** [comment]
Thanks. I will update the README as part of this PR and merge it when I am done. Will do first thing tomorrow.

**Comment by pahud** [comment]
Hooray! Deep breath :-)

**Pull request merged**